draft-ietf-jose-hpke-encrypt.md

```diff
@@ -433,16 +436,15 @@ In Key Agreement with Key Wrapping mode, CEK has to be randomly generated and it
 
 ## HPKE authentication
 
-Authenticated HPKE modes MUST NOT be used with Key Encryption, as this is trivially insecure.
+Authenticated HPKE modes MUST NOT be used for Key Encryption, as the message is not authenticated. Any recipient could act as a man-in-the-middle (MitM) and modify the message.
 
 ## Key Management
 
-A single key SHOULD NOT be used as both authentication key and recipient key, as this has not been proven to be safe.
+A single key MUST NOT be used in both sender and recipient roles. Avoiding the use of the same key for both sender and recipient roles ensures clear cryptographic boundaries and minimizes unintended interactions.
 
-A single key SHOULD NOT be used with both JOSE-HPKE and other algorithms as this might enable cross-protocol attacks.
+A single key MUST NOT be used with both JOSE-HPKE and other algorithms as this might enable cross-protocol attacks.
```

Comment:
For implementations tracking key use (to enforce this and the previous requirement), I think it would be useful to register key uses for HPKE receiver and sender roles in the "JSON Web Key Use" registry.

Reply:
Agreed, added text, see https://github.com/ietf-wg-jose/draft-ietf-jose-hpke-encrypt/commit/d07b510a64e9ecd325f15858f838c24e073cafd8.

Reply:
I would move this registration to separate PR, and include examples